### DISCUSSION TITLE
Interesting

### DISCUSSION THREAD

**Discussion by peterpants2** [D2klaas/Godot-4-VectorFieldNavigation#1] [Ideas]
Saw this on the Godot forums. Cool project. You mention that the asset is unable to handle 3d navigation, ramps/bridges/multi-level buildings/etc. is it possible to use this vectorfield along with the godot Navigation system? Perhaps having the vectorfield for large outdoor areas and then have the AI switch to using the built-in Navmesh for navigating buildings etc.

**Comment by D2klaas** [comment]
Hi,
the algorythm can handle ramps and bridges, but it cant handle way's on top of each others. So it only handles heightmaps (for now).
Using this in combination with navmesh is totaly fine. You only have to mark the cells under the building as unwalkable and place targets into the buildings entrances if needed. Enclose the building in an area, when a entity enters this area (through the set targets) just disable the vfn processing in the entity and calculate a navmesh path to follow.

_The algorythm is totaly able to calculate true 3D solutions, its merely a problem with finding the currently occupied cell in 3D space. When the "2d" variant is stable i maybe will extend this to full 3D._
